### PR TITLE
feat: add tox args and python-version config support

### LINKED ISSUE #1093: feat: add tox-args support for custom CLI flags
## Problem

Currently the `tox` configuration only maps branches to test environments. There's no way for users to pass custom CLI flags to tox (e.g., `-p` for parallel, `-x` for fail-fast).

## Proposed Solution

Add a `tox-args` configuration key:

```yaml
tox:
  main: all
tox-args: "-p -x"
```

## Requirements

- [ ] Add `tox-args` to config schema (`webhook_server/config/schema.yaml`)
- [ ] Read `tox-args` in `github_api.py` `_repo_data_from_config()`
- [ ] Pass args to tox command in `runner_handler.py` `run_tox()`
- [ ] Add tests for new functionality
- [ ] Update example config

### PR DESCRIPTION
**Pull request by rnetser**
## Summary

- Adds `args` key under `tox` config for passing custom CLI flags to tox
- Adds `python-version` key under `tox` config (replaces standalone `tox-python-version`)
- Standalone `tox-python-version` still works but logs a deprecation warning
- Schema enforces `args` and `python-version` as strings via explicit properties

### Config format

```yaml
tox:
  main: all
  args: "-p -x"
  python-version: "3.11"
```

## Files Changed

- `webhook_server/config/schema.yaml` — explicit properties for `args` and `python-version`
- `webhook_server/libs/github_api.py` — pop args/python-version from tox dict, deprecation warning
- `webhook_server/libs/handlers/runner_handler.py` — append tox_args to command
- `webhook_server/tests/` — tests for new config, deprecation, backward compat
- `examples/.github-webhook-server.yaml` — updated example

Closes #1093

Assisted-by: Claude <noreply@anthropic.com>

## Test plan

- [x] All 1568 tests pass
- [x] Schema validates `args` and `python-version` as strings
- [x] `python-version` under tox takes priority over standalone `tox-python-version`
- [x] Deprecation warning logged when standalone key used
- [x] `args` and `python-version` popped from tox dict (not treated as branch names)

## Summary by CodeRabbit

* **New Features**
  * Repositories can now specify tox command-line arguments and Python version under a nested tox configuration; configured args are appended to tox runs.

* **Chores**
  * Top-level tox-python-version is deprecated and emits a deprecation warning.

* **Bug Fixes / Validation**
  * Schema now enforces tox.args as a string to prevent invalid types.

* **Tests**
  * Added tests covering nested tox args/python-version parsing, deprecation behavior, and command construction.

[Review Change Stack](https://app.coderabbit.ai/change-stack/myk-org/github-webhook-server/pull/1094?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: webhook_server/libs/github_api.py [hunk 1]
```diff
@@ -798,7 +798,13 @@ def _repo_data_from_config(self, repository_config: dict[str, Any]) -> None:
...
         _tox = self.config.get_value(value="tox", extra_dict=repository_config)
         self.tox: dict[str, str] = _tox if isinstance(_tox, dict) else {}
-        self.tox_python_version: str = self.config.get_value(value="tox-python-version", extra_dict=repository_config)
+        self.tox_args: str = self.tox.pop("args", "")
+        _tox_python_version_nested = self.tox.pop("python-version", "")
+        _tox_python_version_legacy = self.config.get_value(value="tox-python-version", extra_dict=repository_config)
+        self.tox_python_version: str = _tox_python_version_nested or _tox_python_version_legacy
+
+        if not _tox_python_version_nested and _tox_python_version_legacy:
+            self.logger.warning("'tox-python-version' is deprecated, use 'python-version' under 'tox' instead")
```
[the remark this answers is not in the document]
Comment (rnetser) [reply webhook_server/libs/github_api.py]: Fixed: shallow copy of tox dict to avoid mutating shared config. Used key-presence check instead of truthiness for python-version precedence. Added tests for both edge cases.

File: webhook_server/tests/test_repo_data_from_config.py [hunk 2]
```diff
@@ -19,3 +26,101 @@ def test_repo_data_from_config_repository_found(process_github_webhook):
...
+    assert process_github_webhook.tox_python_version == "3.8"
+    assert "deprecated" not in caplog.text.lower()
+
+
+def test_tox_python_version_legacy_deprecation_warning(process_github_webhook, caplog):
+    """When standalone 'tox-python-version' is used instead of nested 'tox.python-version',
+    a deprecation warning should be logged."""
+    original_get_value = process_github_webhook.config.get_value
+
+    def patched_get_value(value, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (rnetser) [reply webhook_server/tests/test_repo_data_from_config.py]: Fixed: added type hints `(value: str, *args: Any, **kwargs: Any) -> Any` to all 4 `patched_get_value` helpers and added `from typing import Any` import.

**Comment by rnetser** [comment]
@coderabbitai review

File: webhook_server/tests/test_repo_data_from_config.py [hunk 3]
```diff
@@ -19,3 +27,101 @@ def test_repo_data_from_config_repository_found(process_github_webhook):
...
     assert process_github_webhook.minimum_lgtm == 0
+
+
+def test_tox_python_version_nested_no_deprecation_warning(process_github_webhook, caplog):
+    """When 'python-version' is set under 'tox', no deprecation warning should be logged."""
+    with caplog.at_level(python_logging.WARNING):
+        process_github_webhook._repo_data_from_config(repository_config={})
+
+    assert process_github_webhook.tox_python_version == "3.8"
+    assert "deprecated" not in caplog.text.lower()
```
[the remark this answers is not in the document]
Comment (rnetser) [reply webhook_server/tests/test_repo_data_from_config.py]: Fixed: replaced broad caplog.text checks with targeted caplog.records assertions at all 4 locations (lines 38, 60-61, 83, 127).

**Comment by rnetser** [comment]
@coderabbitai review
